In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv(r'data/state-weights.csv')

In [3]:
abbrevs = df["Abbreviation"]
states = df["State"]
weights = df["Cast Ballot Weight"]

# Cast Ballot Weight Comparison Table
A vote cast in Alaska has 1.61 times more power than one cast in California before the results are known

In [5]:
voteWeightsOrdered = [list(i / j for j in weights) for i in weights]
table = pd.DataFrame(voteWeightsOrdered, columns=list(abbrevs), index=list(states))
table

AL        AK        AZ        AR        CA  \
Alabama               1.000000  0.414577  0.752027  0.782688  0.683572   
Alaska                2.412095  1.000000  1.813959  1.887917  1.648841   
Arizona               1.329740  0.551280  1.000000  1.040771  0.908974   
Arkansas              1.277649  0.529684  0.960826  1.000000  0.873366   
California            1.462903  0.606486  1.100142  1.144996  1.000000   
Colorado              0.863746  0.358090  0.649560  0.676043  0.590433   
Connecticut           1.077828  0.446843  0.810555  0.843602  0.736773   
Delaware              1.604537  0.665205  1.206654  1.255851  1.096817   
District of Columbia  2.283864  0.946839  1.717527  1.787552  1.561187   
Florida               0.717725  0.297553  0.539749  0.561755  0.490617   
Georgia               0.921681  0.382108  0.693129  0.721389  0.630036   
Hawaii                2.274794  0.943078  1.710705  1.780453  1.554986   
Idaho                 1.507858  0.625124  1.133949  1.180181  1.030730   
Illinois              0.883105  0.366116  0.664119  0.691196  0.603666   
Indiana               0.956277  0.396451  0.719146  0.748466  0.653685   
Iowa                  0.931223  0.386064  0.700304  0.728857  0.636558   
Kansas                1.260749  0.522678  0.948117  0.986772  0.861813   
Kentucky              0.982657  0.407388  0.738984  0.769114  0.671717   
Louisiana             0.919526  0.381215  0.691508  0.719702  0.628563   
Maine                 1.304900  0.540982  0.981320  1.021329  0.891994   
Maryland              0.948732  0.393323  0.713472  0.742561  0.648527   
Massachusetts         0.812055  0.336660  0.610687  0.635585  0.555098   
Michigan              0.791807  0.328265  0.595460  0.619738  0.541257   
Minnesota             0.836583  0.346828  0.629133  0.654783  0.571865   
Mississippi           1.183740  0.490752  0.890204  0.926499  0.809172   
Missouri              0.852069  0.353249  0.640779  0.666904  0.582451   
Montana               1.504550  0.623753  1.131462  1.177593  1.028469   
Nebraska              1.480899  0.613947  1.113675  1.159081  1.012302   
Nevada                1.286857  0.533502  0.967751  1.007207  0.879660   
New Hampshire         1.297352  0.537853  0.975643  1.015421  0.886834   
New Jersey            0.900858  0.373475  0.677469  0.705090  0.615801   
New Mexico            1.614997  0.669541  1.214520  1.264038  1.103967   
New York              0.962221  0.398915  0.723615  0.753118  0.657747   
North Carolina        0.749724  0.310819  0.563812  0.586800  0.512491   
North Dakota          2.179827  0.903707  1.639288  1.706124  1.490069   
Ohio                  0.795840  0.329937  0.598493  0.622894  0.544014   
Oklahoma              1.151555  0.477409  0.866000  0.901308  0.787171   
Oregon                0.939105  0.389332  0.706232  0.735026  0.641946   
Pennsylvania          0.781573  0.324022  0.587764  0.611727  0.534262   
Rhode Island          2.286352  0.947870  1.719397  1.789499  1.562887   
South Carolina        1.016020  0.421219  0.764074  0.795226  0.694523   
South Dakota          1.955720  0.810797  1.470753  1.530718  1.336876   
Tennessee             1.037953  0.430312  0.780568  0.812393  0.709516   
Texas                 1.000074  0.414608  0.752082  0.782745  0.683623   
Utah                  2.258049  0.936136  1.698113  1.767347  1.543540   
Vermont               2.471027  1.024432  1.858278  1.934042  1.689126   
Virginia              0.801813  0.332413  0.602984  0.627569  0.548097   
Washington            1.312712  0.544221  0.987194  1.027443  0.897334   
West Virginia         1.670002  0.692345  1.255886  1.307090  1.141567   
Wisconsin             0.805971  0.334137  0.606111  0.630823  0.550939   
Wyoming               2.932284  1.215659  2.205156  2.295062  2.004428   

                            CO        CT        DE        DC        FL  \
Alabama               1.157747  0.927792  0.623233  0.437854  1.393290   
Alaska                2.792597 

In [6]:
abbrevs = df["Abbreviation"]
states = df["State"]
weights = df["Final Ballot Weight"]

# Final Ballot Weight Comparison Table
A vote cast in California for the winning candidate had 1.46 times more power than one cast for the winning candidate in Alabama. We know this after the results of the election are known

In [8]:
voteWeightsOrdered = [list(i / j for j in weights) for i in weights]
table = pd.DataFrame(voteWeightsOrdered, columns=list(abbrevs), index=list(states))
table

AL        AK        AZ        AR        CA  \
Alabama               1.000000  0.375049  0.609070  0.778052  0.687234   
Alaska                2.666318  1.000000  1.623975  2.074535  1.832384   
Arizona               1.641847  0.615773  1.000000  1.277443  1.128333   
Arkansas              1.285261  0.482036  0.782814  1.000000  0.883275   
California            1.455108  0.545737  0.886263  1.132150  1.000000   
Colorado              1.081806  0.405731  0.658896  0.841702  0.743454   
Connecticut           1.234572  0.463025  0.751941  0.960562  0.848440   
Delaware              1.849222  0.693549  1.126306  1.438792  1.270849   
District of Columbia  1.540295  0.577686  0.938148  1.198430  1.058543   
Florida               0.914383  0.342938  0.556923  0.711437  0.628395   
Georgia               1.123173  0.421245  0.684091  0.873888  0.771883   
Hawaii                2.176899  0.816444  1.325885  1.693742  1.496039   
Idaho                 1.426466  0.534995  0.868818  1.109865  0.980316   
Illinois              0.975409  0.365826  0.594092  0.758919  0.670334   
Indiana               1.026431  0.384962  0.625169  0.798617  0.705398   
Iowa                  1.090572  0.409018  0.664235  0.848522  0.749478   
Kansas                1.328158  0.498124  0.808941  1.033376  0.912755   
Kentucky              0.965725  0.362194  0.588194  0.751385  0.663679   
Louisiana             0.986169  0.369862  0.600646  0.767291  0.677729   
Maine                 1.636798  0.613880  0.996925  1.273515  1.124864   
Maryland              0.969417  0.363579  0.590443  0.754257  0.666216   
Massachusetts         0.812998  0.304914  0.495173  0.632555  0.558720   
Michigan              1.019398  0.382324  0.620885  0.793145  0.700565   
Minnesota             1.062551  0.398509  0.647168  0.826720  0.730221   
Mississippi           1.284213  0.481643  0.782176  0.999185  0.882555   
Missouri              0.915741  0.343448  0.557751  0.712494  0.629328   
Montana               1.589237  0.596042  0.967957  1.236510  1.092178   
Nebraska              1.494527  0.560521  0.910272  1.162820  1.027090   
Nevada                1.620230  0.607666  0.986834  1.260623  1.113477   
New Hampshire         1.668521  0.625777  1.016246  1.298197  1.146664   
New Jersey            1.026885  0.385132  0.625445  0.798970  0.705710   
New Mexico            1.907076  0.715247  1.161543  1.483805  1.310607   
New York              1.016329  0.381173  0.619016  0.790757  0.698456   
North Carolina        0.931016  0.349177  0.567054  0.724379  0.639826   
North Dakota          2.015618  0.755956  1.227653  1.568257  1.385202   
Ohio                  0.942953  0.353654  0.574324  0.733667  0.648029   
Oklahoma              1.072329  0.402176  0.653124  0.834328  0.736941   
Oregon                1.087592  0.407900  0.662420  0.846203  0.747430   
Pennsylvania          0.997026  0.373934  0.607259  0.775738  0.685190   
Rhode Island          2.558249  0.959469  1.558153  1.990452  1.758116   
South Carolina        1.142806  0.428608  0.696049  0.889162  0.785375   
South Dakota          1.913218  0.717551  1.165284  1.488584  1.314829   
Tennessee             1.052689  0.394810  0.641162  0.819047  0.723444   
Texas                 1.178687  0.442065  0.717903  0.917080  0.810034   
Utah                  2.323385  0.871383  1.415105  1.807715  1.596709   
Vermont               2.445817  0.917302  1.489675  1.902974  1.680849   
Virginia              0.984837  0.369362  0.599835  0.766255  0.676814   
Washington            1.426655  0.535066  0.868933  1.110012  0.980446   
West Virginia         1.493362  0.560084  0.909562  1.161913  1.026289   
Wisconsin             1.030275  0.386404  0.627510  0.801608  0.708040   
Wyoming               2.500124  0.937669  1.522751  1.945227  1.718171   

                            CO        CT        DE        DC        FL  \
Alabama               0.924380  0.809997  0.540768  0.649226  1.093634   
Alaska                2.464690 